### ISSUE TITLE
Bad localization for names in links

### ISSUE THREAD

**Issue by Bibi56** [openstreetmap/openstreetmap-website#2755]
[Freiburg Minster](https://www.openstreetmap.org/way/110404213) has several names:

tag | value
-- | --
name 	 |Münster Unserer Lieben Frau
name:de 	 |Münster Unserer Lieben Frau
name:en 	 |Freiburg Minster
name:he 	 |קתדרלת פרייבורג
name:ru 	 |Фрайбургский мюнстер

[Wermter Fashion](https://www.openstreetmap.org/node/6352395085) has two names:

tag | value
-- | --
name  |Wermter Mode
name:en  |Wermter Fashion

Both are located in [Germany](https://www.openstreetmap.org/relation/51477) for which the official language is German `default_language=de`).

Let's set `fr` as preferred language in user preference if the user is connected (or in the browser preference if not connected).

For the minster I get:
Chemin : Freiburg Minster (110404213).

_Chemin_, not Weg or Way as I said I prefer French user interface. That's good.

But _Freiburg Minster_ instead of _Münster Unserer Lieben Frau_ :-(.
Why not using the default name as I didn't say I spoke English? The map by default displays the local name.
As German and French share a Latin alphabet (like English) there is no good reason to promote English over the local name, is it? A bad one may be simplicity^^.

I would expect: use `name:fr` if it exists, else use `int_name`, else use `name`.

For the fashion shop I get:
Nœud : Wermter Fashion (6352395085).

Same issue.

If I set the locales to `fr de`, then I get:
Chemin : Münster Unserer Lieben Frau (110404213). Good but for a bad reason (the German name is entered twice, once as `name` as it should but a second - redundant - time as `name:de`).
Nœud : Wermter Fashion (6352395085).
This time bad for two reasons: 

- I'm still not supposed to speak English, therefore name is better.
- I'm supposed to speak German and name is in German (due to the location).

Note: ticket due to the [discussion on OSM France mailing list](http://gis.19327.n8.nabble.com/noms-anglais-en-Allemagne-td5972300.html).

**Comment by tomhughes** [comment]
We prefer the user's language because they're the one reading the page.

At least I think that's what you're asking - it's a bit hard to tell in all the text above what the actual question is.

**Comment by Bibi56** [comment]
The request is to use the preferred language and the issue is that for the object name it is _not_ the case.
`name:en`=_Freiburg Minster_ is promoted to French speaking people. It should not.
`name:en`=_Wermter Fashion_ is promoted to German speaking people. It should not.

**Comment by tomhughes** [comment]
I get `Chemin : Freiburg Minster` when my locale is `fr` and `Weg: Münster Unserer Lieben Frau` when my locale is `de` which seems right?

**Comment by Bibi56** [comment]
No, `Chemin : Münster Unserer Lieben Frau` would be the correct "French" version (as long as nobody enters `name:fr=Cathédrale Notre-Dame` of course).
`Weg: Münster Unserer Lieben Frau` is correct but for a bad reason (duplication of name into `name:de`). The error is then visible on `Wermter Fashion`.

**Comment by tomhughes** [comment]
Ah actually that was probably because I have `en` in my browser languages so it picked that up when there wasn't a french name.

**Comment by tomhughes** [comment]
But really I spent ages on this not so long ago and I no real wish to try and change it again - every time I improve it for one person I make it worse for somebody else, at least according to each person's personal theory of what is "right".

**Comment by tomhughes** [comment]
Actually I suspect it's precisely because the last person to moan about this made me add the full language fallback chain for the current user to the selection list and that will always have english at the end.

That may be quite hard to fix unfortunately...

**Comment by Bibi56** [comment]
> Ah actually that was probably because I have en in my browser languages so it picked that up when there wasn't a french name.

No, even if you remove `en` from your browser languages. I did check with/without being connected, Browser preferences are used only when the user is not connected.

OSM should be neutral. OSM default map is neutral.
What I do expect is that the "known" user languages are used.
And here English is used for no good reasons. If a user wants to see English he has to tune his/her account or his/her browser settings.

> Don't fall back to name:en if it's not in the user's language list

Sounds good to me.

One tricky part is to know the language of name. Here, thanks to the position and the default_language=de we know that name means name:de.
Apparently (see _Wermter Fashion_ example), it's not the case. You may want to create a specific issue.

So use `name` if `de` is in the user preference list or as very fail-back.

I was mentioning the alphabet as getting something in Cyrillic or Arabic alphabet may be an issue for some users. But then transliteration is probably better (openstreetmap.de does the job). I'm not saying it should be done here.

**Comment by dieterdreist** [comment]
sent from a phone

>> On 6. Aug 2020, at 14:15, Bibi56 <notifications@github.com> wrote:
> I would expect: use name:fr if it exists, else use int_name, else use name.


I would depreciate int_name there is no such thing like a (single) international name, it is just another way of saying English or English transliteration.

I would also question that a local clothing shop in Germany has an English name, save very few exceptions, nor that it is useful to translate a German name and store the result in OpenStreetMap. Indeed "Wermter Fashion" only yields results that are based on OpenStreetMap or where the words are coupled by coincidence (Dunja Wermter Fashion Show).

I would not be very surprised if this  derives from an iD feature, for example I have recently found a local bank which had an otherwise non-existing English brand name (translation) and the reason was in iD’s name suggestion index (even misspelled)

https://www.openstreetmap.org/changeset/81580509#map=15/41.8608/12.4878

Cheers Martin

**Comment by Bibi56** [comment]
> I would depreciate int_name there is no such thing like a (single) international name, it is just another way of saying English or English transliteration.

Not necessary, on some subjects the international name is French. But we can skip it as in most usage it's currently BS, Top usage for thousands of [Aparato Individual de Bombeo](https://taginfo.openstreetmap.org/keys/int_name#values).
And no, Individual Pumping Device is not an international name but a description. BTW I just commented one changeset ;-).

> I would also question 

Me too, but it's just an example, I used Overpass to find an example in Germany with just `name` and `name:en`, without (IMHO useless) `name:de`.

I do question the usage of `name:de` in Germany as the default language in Germany is German. With some possible exceptions like a bilingual name as name but split for instance in `name:de` and `name:gsw`.

> Cheers Martin

Martin, the international name or the English name of Dieter? ;-)
